// vuepress-theme-hope@2.0.0-beta.87/lib/node/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sitemap = exports.seo = exports.readingTime = exports.pwa = exports.photoSwipe = exports.mdEnhance = exports.git = exports.feed = exports.copyright = exports.copyCode = exports.comment = exports.blog = exports.objectSidebar = exports.arraySidebar = exports.sidebar = exports.navbar = void 0;
const navbar = (navbarConfig) => navbarConfig;
exports.navbar = navbar;
const sidebar = (sidebarConfig) => sidebarConfig;
exports.sidebar = sidebar;
const arraySidebar = (sidebarConfig) => sidebarConfig;
exports.arraySidebar = arraySidebar;
const objectSidebar = (sidebarConfig) => sidebarConfig;
exports.objectSidebar = objectSidebar;
const blog = (blogConfig) => blogConfig;
exports.blog = blog;
const comment = (commentConfig) => commentConfig;
exports.comment = comment;
const copyCode = (copyCodeConfig) => copyCodeConfig;
exports.copyCode = copyCode;
const copyright = (copyrightConfig) => copyrightConfig;
exports.copyright = copyright;
const feed = (feedConfig) => feedConfig;
exports.feed = feed;
const git = (gitConfig) => gitConfig;
exports.git = git;
const mdEnhance = (mdEnhanceConfig) => mdEnhanceConfig;
exports.mdEnhance = mdEnhance;
const photoSwipe = (photoSwipeConfig) => photoSwipeConfig;
exports.photoSwipe = photoSwipe;
const pwa = (pwaConfig) => pwaConfig;
exports.pwa = pwa;
const readingTime = (readingTimeConfig) => readingTimeConfig;
exports.readingTime = readingTime;
const seo = (seoConfig) => seoConfig;
exports.seo = seo;
const sitemap = (sitemapConfig) => sitemapConfig;
exports.sitemap = sitemap;
